models/aeif_cond_beta_multisynapse.h

```diff
@@ -0,0 +1,401 @@
+    double a;       //!< Subthreshold adaptation in nS.
+    double b;       //!< Spike-triggered adaptation in pA
+    double V_th;    //!< Spike threshold in mV.
+    double t_ref;   //!< Refractory period in ms.
+    std::vector< double > taus_rise;  //!< Rise time of synaptic conductance
+                                      //!< in ms..
+    std::vector< double > taus_decay; //!< Decay time of synaptic conductance
+                                      //!< in ms..
+
+    double I_e; //!< Intrinsic current in pA.
+
+    double gsl_error_tol; //!< error bound for GSL integrator
+
+    // type is long because other types are not put through in GetStatus
+    std::vector< long > receptor_types_;
+    size_t num_of_receptors_;
+
+    // boolean flag which indicates whether the neuron has connections
+    bool has_connections_;
+
+    Parameters_(); //!< Sets default parameter values
+
+    void get( DictionaryDatum& ) const; //!< Store current values in dictionary
+    void set( const DictionaryDatum& ); //!< Set values from dictionary
+  };
+
+  // ----------------------------------------------------------------
+
+  /**
+   * State variables of the model.
+   * @note Copy constructor and assignment operator required because
+   *       of C-style arrays.
+   */
+  struct State_
+  {
+
+    /**
+     * Enumeration identifying elements in state vector State_::y_.
+     * This enum identifies the elements of the vector. It must be public to be
+     * accessible from the iteration function. The last four elements of this
+     * enum (DG_EXC, G_EXC, DG_INH, G_INH) will be repeated
+     * n times at the end of the state vector State_::y with n being the number
+     * of synapses.
+     */
+    enum StateVecElems
+    {
+      V_M = 0,
+      W,      // 1
+      DG_EXC, // 2
+      G_EXC,  // 3
+      DG_INH, // 4
+      G_INH,  // 5
+      STATE_VECTOR_MIN_SIZE
+    };
+
+    static const size_t NUMBER_OF_FIXED_STATES_ELEMENTS = 2; // V_M, W
+    // DG_EXC, G_EXC, DG_INH, G_INH
+    static const size_t NUMBER_OF_STATES_ELEMENTS_PER_RECEPTOR = 4;
+
+    std::vector< double > y_; //!< neuron state
+    int_t r_;                 //!< number of refractory steps remaining
+
+    State_( const Parameters_& ); //!< Default initialization
+    State_( const State_& );
+    State_& operator=( const State_& );
+
+    void get( DictionaryDatum& ) const;
+    void set( const DictionaryDatum& );
+
+  }; // State_
+
+  // ----------------------------------------------------------------
+
+  /**
+   * Buffers of the model.
+   */
+  struct Buffers_
+  {
+    Buffers_( aeif_cond_beta_multisynapse& );
+    Buffers_( const Buffers_&, aeif_cond_beta_multisynapse& );
+
+    //! Logger for all analog data
+    UniversalDataLogger< aeif_cond_beta_multisynapse > logger_;
+
+    /** buffers and sums up incoming spikes/currents */
+    std::vector< RingBuffer > spike_exc_;
+    std::vector< RingBuffer > spike_inh_;
```

Comment:
Here I need qualified help of @heplesser:

We already discussed the issue with our interpretation of inhibitory and excitatory buffers in the current NEST state. In a model with one buffer (iaf_psc_exp) the logic to use the sigh of the weight to route spike events is at least plausible. In the case of the mutlisynapse model it leads to: a) you always have the same number of inh und exc buffers b) you have to buffers (inh/exc) at the same portnumber c) finally, this implementation is inconsistent with the implementation of the iaf_psc_alpha_multisynapse and iaf_psc_exp_multisynapse.